doc/sphinx/source/backends/dynamicbackend-api/index.rst

```diff
@@ -0,0 +1,9 @@
+.. backends/hosttensor-api/index.rst:
+
+
+DynamicBackend
```

Comment:
I'm a bit on the fence about documenting this here. The thing is, the `DynamicBackend` is an internal implementation detail. It's a wrapper class that is not intended to be instantiated directly by the user (instead it should come from `Backend::create()`, and the fact that a user is talking to a `DynamicBackend` (as opposed to any other kind of `Backend` object) should be transparent. It would be more appropriate to document the behavior of `create_dynamic_tensor` within the `Backend` class. On the other hand, the docstrings in `DynamicBackend` are all we have for now.

Reply:
> On the other hand, the docstrings in DynamicBackend are all we have for now.

That's pretty much why I added it.  Useful info, and probably better than nothing.  I'll leave it for now and feel free to remove any time.